plugins/modules/iosxr_vrf.py

```diff
@@ -0,0 +1,244 @@
+                            type: bool
+              import:
+                description: VRF import
+                type: dict
+                suboptions:
+                  route_policy: *route_policy
+                  route_target: *route_target
+                  from:
+                    description: Import routes from a VRF
+                    type: dict
+                    suboptions:
+                      bridge_domain:
+                        description: VRF import
+                        type: dict
+                        suboptions:
+                          advertise_as_vpn: &advertise_as_vpn
+                            description: Advertise local EVPN imported routes to PEs
+                            type: bool
+                      default_vrf: *default_vrf
+                      vrf:
+                        description: Import routes from a VRF
+                        type: dict
+                        suboptions:
+                          advertise_as_vpn: *advertise_as_vpn
+              maximum:
+                description: Set maximum prefix limit
+                type: dict
+                suboptions:
+                  prefix:
+                    description:  Set table's maximum prefix limit.
+                    type: int
+                  threshold:
+                    description: mid-thresh (% of max).
+                    type: int
+          evpn_route_sync:
+            description: EVPN Instance VPN ID used to synchronize the VRF route(s).
+            type: int
+          fallback_vrf:
+            description: Fallback VRF name
+            type: str
+          mhost:
+            description: Multicast host stack options
+            type: dict
+            suboptions:
+              afi:
+                description: Address Family Identifier (AFI)
+                type: str
+                choices: ['ipv4', 'ipv6']
+              default_interface:
+                description: Default interface for multicast.
+                type: str
+          rd:
+            description: VPN Route Distinguisher (RD).
+            type: str
+          remote_route_filtering:
+            description: Enable/Disable remote route filtering per VRF
+            type: bool
+          vpn:
+            description: VPN ID for the VRF
+            type: dict
+            suboptions:
+              id:
+                description: VPN ID for the VRF.
+                type: int
+  running_config:
+    description: The state the configuration should be left in.
+      - State I(deleted) only removes VRF attributes that this modules
+      manages and does not negate the VRF process completely. Thereby, preserving
+      address_family related configurations under VRF context.
+      - Refer to examples for more details.https://github.com/ansible-network/resource_module_models/pull/243/commits/4c40fc3daa8e484051aac3eddf7c4b420b06dbdb
```

Comment:
this is not an expected way to put it! 
Please follow how other modules handle it, and also add the show running configuration command used.